clients/drcachesim/tracer/output.cpp

```diff
@@ -1426,6 +1426,21 @@ exit_thread_io(void *drcontext)
          (!has_tracing_windows() && align_attach_detach_endpoints()));
 
     if (has_prior_window_data || is_not_empty) {
+        if (op_offline.get_value()) {
+            // Provide the PC in case detach exited mid-block.
+            dr_mcontext_t mc;
+            mc.size = sizeof(mc);
+            mc.flags = DR_MC_CONTROL;
+            if (dr_get_mcontext(drcontext, &mc)) {
+                NOTIFY(2, "Thread " TIDFMT "%d end pc=%p\n", dr_get_thread_id(drcontext),
+                       mc.pc);
+                // XXX i#5790: This same marker can be used to solve block truncation for
+                // DR mid-block relocation once DR provides a relocation event.
```

Comment:
Re the asynchronous thread relocation event: is that specific to only languages where the runtime supports it, like Rust? Do we expect it in C++? Is there a high level issue filed for Rust support?

Reply:
Oh no, that's when DR relocates a thread (can happen for several reasons: see #5790).